### PR TITLE
Data Prepper ToC Update

### PR DESCRIPTION
**Pull request by carolxob**
Signed-off-by: carolxob <carolxob@amazon.com>

### Description
Modifies Data Prepper documentation table of contents for accuracy and usability.

### Issues Resolved
Closes issue #2453.

Modifies existing ToC to look like the following:
```Data Prepper
Getting started
Using for common scenarios
  - Trace analytics [pending review]
  - Log analytics
  - S3 log analytics [removed until content exists]
  - Metrics ingestion [removed until content exists]
Pipelines
  - Pipeline options [Placeholder, content moved to Authoring pipelines page]
  - Expression syntax
  - Sources
  - Buffers
  - Processors
  - Sinks
Administrating Data Prepper
  - Configuring Data Data Prepper
  - Peer forwarder [pending review]
  - Core APIs
  - Log4j configuration
  - Monitoring Data Prepper
Migrating from Logstash
Migrating from Open Distro Data Prepper
```

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by carolxob** [comment]
@dlvenable Where should the Pipelines section fit in this structure? Right now, it's showing as last in the TOC, but can be moved to wherever it should fit.

**Comment by dlvenable** [comment]
> @dlvenable Where should the Pipelines section fit in this structure? Right now, it's showing as last in the TOC, but can be moved to wherever it should fit.

Thanks for catching that. I can see two good places for it:

1. At the top-level.

```
Data Prepper
Getting started
Pipelines
Using for common scenarios
...
```

2. Under the "Authoring pipelines" section.

```
Authoring Pipelines
  - Pipelines
  - Pipeline options
  ...
```

@carolxob , Which do you think would be the best location? The concept of pipelines is very core to Data Prepper, so I tend to think option #1.

**Comment by carolxob** [comment]
@dlvenable Regarding Pipelines, combined both of your suggestions so that it looks like this:
```Pipelines
       -Pipeline options
       -Expression syntax
       - Sources
       - Buffers
       - Processors
       - Sinks
```

How does this look to you?

File: _data-prepper/migrating-from-logstash-data-prepper.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+redirect_from: 
   - /data-prepper/configure-logstash-data-prepper/
 ---
 
 # Migrating from Logstash
 
 You can run Data Prepper with a Logstash configuration.
 
-As mentioned in the [Getting started]({{site.url}}{{site.baseurl}}/data-prepper/get-started/) guide, you'll need to configure Data Prepper with a pipeline using a `pipelines.yaml` file.
+As mentioned in the [Getting started]({{site.url}}{{site.baseurl}}/data-prepper/getting-started/) guide, you'll need to configure Data Prepper with a pipeline using a `pipelines.yaml` file.
```
Comment (hdhalter) [inline _data-prepper/migrating-from-logstash-data-prepper.md:13]: guide --> topic?

**Review by hdhalter — COMMENTED** [review]
Thanks, Caroline, just a few catches.

**Comment by hdhalter** [comment]
I think there's a usage of S3 Source and should be S3Source.

File: _data-prepper/administrating-data-prepper/administrating-data-prepper.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+layout: default
+title: Administrating Data Prepper
+has_children: true
+nav_order: 20
+---
+
+# Administrating Data Prepper
```
[the remark this answers is not in the document]
Comment (dlvenable) [reply _data-prepper/administrating-data-prepper/administrating-data-prepper.md:8]: I understood that we want want to start with a gerund. If so, then "Managing Data Prepper" would work. If it doesn't matter, I'm fine with "Data Prepper Administration" as well.

File: _data-prepper/migrating-from-logstash-data-prepper.md (hunk 1 above)
Comment (natebower) [reply _data-prepper/migrating-from-logstash-data-prepper.md:13]: Let's avoid this by instead saying "As mentioned in [Getting started with Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/getting-started/), you'll need..."

File: _data-prepper/getting-started.md [hunk 3]
```diff
@@ -71,7 +69,7 @@ docker run --name data-prepper \
     opensearchproject/data-prepper:latest
 ```
 
-This sample pipeline configuration above demonstrates a simple pipeline with a source (`random`) sending data to a sink (`stdout`). For more examples and details about more advanced pipeline configurations, see [Pipelines]({{site.url}}{{site.baseurl}}/clients/data-prepper/pipelines).
+This sample pipeline configuration above demonstrates a simple pipeline with a source (`random`) sending data to a sink (`stdout`). For more examples and details about more advanced pipeline configurations, see [Pipelines]({{site.url}}{{site.baseurl}}/clients/data-prepper/pipelines/).
```
Comment (kolchfa-aws) [inline _data-prepper/getting-started.md:72]:
```suggestion
The preceding sample pipeline configuration demonstrates a simple pipeline with a source (`random`) sending data to a sink (`stdout`). For further detailed examples of more advanced pipeline configurations, see [Pipelines]({{site.url}}{{site.baseurl}}/clients/data-prepper/pipelines/).
```

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md [hunk 4]
```diff
@@ -31,15 +29,15 @@ peer_forwarder | No | Object | Peer forwarder configurations. See [Peer forwarde
...
+#### General options for peer forwarding
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
-port | No | Integer | The port number peer forwarder server is running on. Valid options are between 0 and 65535. Defaults is 4994.
-request_timeout | No | Integer | Request timeout in milliseconds for peer forwarder HTTP server. Default is 10000.
-server_thread_count | No | Integer | Number of threads used by peer forwarder server. Default is 200.
-client_thread_count | No | Integer | Number of threads used by peer forwarder client. Default is 200.
-max_connection_count | No | Integer | Maximum number of open connections for peer forwarder server. Default is 500.
+port | No | Integer | The port number that the peer forwarding server is running on. Valid options are between 0 and 65535. Defaults is 4994.
```
Comment (kolchfa-aws) [inline _data-prepper/managing-data-prepper/configuring-data-prepper.md:36]:
```suggestion
port | No | Integer | The peer forwarding server port. Valid options are between 0 and 65535. Defaults is 4994.
```

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md [hunk 5]
```diff
@@ -31,15 +29,15 @@ peer_forwarder | No | Object | Peer forwarder configurations. See [Peer forwarde
...
+request_timeout | No | Integer | Request timeout in milliseconds for the peer forwarder HTTP server. Default is 10000.
```
Comment (kolchfa-aws) [inline _data-prepper/managing-data-prepper/configuring-data-prepper.md:37]:
```suggestion
request_timeout | No | Integer | Request timeout for the peer forwarder HTTP server, in milliseconds. Default is 10000.
```

File: _data-prepper/migrating-from-logstash-data-prepper.md [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
+redirect_from: 
   - /data-prepper/configure-logstash-data-prepper/
 ---
 
 # Migrating from Logstash
 
 You can run Data Prepper with a Logstash configuration.
 
-As mentioned in the [Getting started]({{site.url}}{{site.baseurl}}/data-prepper/get-started/) guide, you'll need to configure Data Prepper with a pipeline using a `pipelines.yaml` file.
+As mentioned in the [Getting started with Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/getting-started/), you'll need to configure Data Prepper with a pipeline using a `pipelines.yaml` file.
```
Comment (kolchfa-aws) [inline _data-prepper/migrating-from-logstash-data-prepper.md:13]:
```suggestion
As mentioned in [Getting started with Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/getting-started/), you'll need to configure Data Prepper with a pipeline using a `pipelines.yaml` file.
```

File: _data-prepper/pipelines/configuration/sources/s3.md [hunk 7]
```diff
@@ -20,14 +20,14 @@ codec | Yes | Codec | The codec to apply. Must be `newline`, `json`, or `csv`.
...
+buffer_timeout | No | Duration | The timeout for writing events to the Data Prepper buffer. Any events that the S3Source cannot write to the buffer in this time will be discarded. Default is 10 seconds.
 records_to_accumulate | No | Integer | The number of messages that accumulate before writing to the buffer. Default is 100.
 metadata_root_key | No | String | Base key for adding S3 metadata to each Event. The metadata includes the key and bucket for each S3 object. Defaults to `s3/`.
-disable_bucket_ownership_validation | No | Boolean | If `true`, then the S3 Source will not attempt to validate that the bucket is owned by the expected account. The only expected account is the same account that owns the SQS queue. Defaults to `false`.
+disable_bucket_ownership_validation | No | Boolean | If `true`, then the S3Source will not attempt to validate that the bucket is owned by the expected account. The only expected account is the same account that owns the SQS queue. Defaults to `false`.
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/s3.md:26]:
```suggestion
disable_bucket_ownership_validation | No | Boolean | If `true`, the S3Source will not attempt to validate that the bucket is owned by the expected account. The expected account is the same account that owns the SQS queue. Defaults to `false`.
```

File: _data-prepper/pipelines/configuration/sources/s3.md [hunk 8]
```diff
@@ -20,14 +20,14 @@ codec | Yes | Codec | The codec to apply. Must be `newline`, `json`, or `csv`.
...
 
 ## sqs
 
-The following are configure usage of Amazon SQS in the S3 Source plugin.
+The following are configure usage of Amazon SQS in the S3Source plugin.
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/s3.md:30]: Could we clarify this sentence? I'm not sure what "configure usage" means...

File: _data-prepper/pipelines/pipelines.md [hunk 9]
```diff
@@ -77,7 +79,12 @@ conditional-routing-sample-pipeline:
 
 ## Examples
 
-This section provides some pipeline examples that you can use to start creating your own pipelines. For more information, see [Data Prepper configuration reference]({{site.url}}{{site.baseurl}}/clients/data-prepper/data-prepper-reference/) guide.
+This section provides some pipeline examples that you can use to start creating your own pipelines. For more options when creating pipelines, select from the following options for each component:
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/pipelines.md:82]:
```suggestion
This section provides some pipeline examples that you can use to start creating your own pipelines. For more pipeline configurations, select from the following options for each component:
```

File: _data-prepper/pipelines/pipelines.md [hunk 10]
```diff
@@ -293,13 +297,13 @@ docker run --name data-prepper \
...
 
 Data Prepper provides an HTTP service to forward Events between Data Prepper nodes for aggregation. This is required for operating Data Prepper in a clustered deployment. Currently, peer forwarding is supported in `aggregate`, `service_map_stateful`, and `otel_trace_raw` processors. Peer forwarder groups events based on the identification keys provided by the processors. For `service_map_stateful` and `otel_trace_raw` it's `traceId` by default and can not be configured. For `aggregate` processor, it is configurable using `identification_keys` option. 
 
-Peer forwarder supports peer discovery through one of three options: a static list, a DNS record lookup , or AWS Cloud Map. This option can be configured using `discovery_mode` option. Peer forwarder also supports SSL for verification and encrytion, and mTLS for mutual authentication in peer forwarding service.
+Peer Forwarder supports peer discovery through one of three options: a static list, a DNS record lookup , or AWS Cloud Map. This option can be configured using `discovery_mode` option. Peer forwarder also supports SSL for verification and encrytion, and mTLS for mutual authentication in peer forwarding service.
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/pipelines.md:304]:
```suggestion
Peer Forwarder supports peer discovery through one of three options: a static list, a DNS record lookup , or AWS Cloud Map. Peer discovery can be configured using `discovery_mode` option. Peer forwarder also supports SSL for verification and encryption, and mTLS for mutual authentication in a peer forwarding service.
```

File: _data-prepper/pipelines/pipelines.md [hunk 11]
```diff
@@ -213,10 +220,7 @@ metrics-pipeline:
 
 ### S3 log ingestion pipeline
 
-The following example demonstrates how to use the S3 Source and Grok Processor plugins to process unstructured log data
-from [Amazon Simple Storage Service](https://aws.amazon.com/s3/) (Amazon S3). This example uses Application Load 
-Balancer logs. As the Application Load Balancer writes logs to S3, S3 creates notifications in Amazon SQS. Data Prepper 
-reads those notifications and reads the S3 objects to get the log data and process it.
+The following example demonstrates how to use the S3Source and Grok Processor plugins to process unstructured log data from [Amazon Simple Storage Service](https://aws.amazon.com/s3/) (Amazon S3). This example uses application load balancer logs. As the application load balancer writes logs to S3, S3 creates notifications in Amazon SQS. Data Prepper reads those notifications and reads the S3 objects to get the log data and process it.
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/pipelines.md:223]:
```suggestion
The following example demonstrates how to use the S3Source and Grok Processor plugins to process unstructured log data from [Amazon Simple Storage Service](https://aws.amazon.com/s3/) (Amazon S3). This example uses application load balancer logs. As the application load balancer writes logs to S3, S3 creates notifications in Amazon SQS. Data Prepper monitors those notifications and reads the S3 objects to get the log data and process it.
```

File: _data-prepper/pipelines/pipelines.md [hunk 12]
```diff
@@ -293,13 +297,13 @@ docker run --name data-prepper \
...
 
-Peer forwarder supports peer discovery through one of three options: a static list, a DNS record lookup , or AWS Cloud Map. This option can be configured using `discovery_mode` option. Peer forwarder also supports SSL for verification and encrytion, and mTLS for mutual authentication in peer forwarding service.
+Peer Forwarder supports peer discovery through one of three options: a static list, a DNS record lookup , or AWS Cloud Map. This option can be configured using `discovery_mode` option. Peer forwarder also supports SSL for verification and encrytion, and mTLS for mutual authentication in peer forwarding service.
 
-To configure the peer forwarder, add configuration options to `data-prepper-config.yaml` mentioned in the previous [Configure the Data Prepper server](#configure-the-data-prepper-server) section:
+To configure peer forwarder, add configuration options to `data-prepper-config.yaml` mentioned in the previous [Configure the Data Prepper server](#configure-the-data-prepper-server) section:
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/pipelines.md:306]:
```suggestion
To configure peer forwarder, add configuration options to `data-prepper-config.yaml` mentioned in the [Configure the Data Prepper server](#configure-the-data-prepper-server) section:
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM. Left some suggestions.

File: _data-prepper/getting-started.md (hunk 3 above)
Comment (vagimeli) [reply _data-prepper/getting-started.md:72]: +1 @kolchfa-aws

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md (hunk 4 above)
Comment (vagimeli) [reply _data-prepper/managing-data-prepper/configuring-data-prepper.md:36]: +1

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md (hunk 5 above)
Comment (vagimeli) [reply _data-prepper/managing-data-prepper/configuring-data-prepper.md:37]: +1

File: _data-prepper/migrating-from-logstash-data-prepper.md [hunk 13]
```diff
@@ -28,7 +28,7 @@ As of the Data Prepper 1.2 release, the following plugins from the Logstash conf
 
 ## Running Data Prepper with a Logstash configuration
 
-1. To install Data Prepper's Docker image, see the Installing Data Prepper in [Get Started]({{site.url}}{{site.baseurl}}/data-prepper/getting-started#1-installing-data-prepper).
+1. To install Data Prepper's Docker image, see the Installing Data Prepper in [Getting Started]({{site.url}}{{site.baseurl}}/data-prepper/getting-started#1-installing-data-prepper).
```
Comment (vagimeli) [inline _data-prepper/migrating-from-logstash-data-prepper.md:31]: Delete "the" before "Installing."

File: _data-prepper/pipelines/configuration/sources/s3.md (hunk 8 above)
Comment (vagimeli) [reply _data-prepper/pipelines/configuration/sources/s3.md:30]: If it helps our understanding of S3Source, see the QuickSight documentation about S3Source https://docs.aws.amazon.com/quicksight/latest/APIReference/API_S3Source.html

**Review by vagimeli — APPROVED** [review]
Minimal comment. Good to go to editorial once doc review feedback is addressed.

File: _data-prepper/migrating-from-logstash-data-prepper.md (hunk 6 above)
Comment (carolxob) [reply _data-prepper/migrating-from-logstash-data-prepper.md:13]: @kolchfa-aws Good catch on this one.

**Pull request merged**